File: LeetCode/2109/solution.py
# 2109. Adding Spaces to a String
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        n, m = len(s), len(spaces)
        result = ""
        j = 0

        for i in range(n):
            if j < m and i == spaces[j]:
                result += ' '
                j += 1
            result += s[i]

        return result

File: LeetCode/2109/solution.cpp
// 2109. Adding Spaces to a String
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string str = "";
        int m=s.size(), n=spaces.size();
        for(int i=0, j=0; i<m; i++){
            if(j<n && i==spaces[j]){
                str += ' ';
                j++;
            }
            str += s[i];
        }
        return str;
    }
};